### PR TITLE
Add dialog before drafting intent email.

### PR DESCRIPTION
**Pull request by dlaliberte**
Add interstitial dialog before drafting email for, e.g. Intent to Prototype, which will be displayed if there are remaining non-email items.

Direct links, if available, to edit the missing items, are provided.

### CODE REVIEW

**Comment by dlaliberte** [comment]
Regarding your suggestion in #1801: "To make your dialog box have "Edit" links that jump straight to the field that needs to be edited, you will need to add logic that loops over the stages looking for the one that has the needed progress item. Then use that stage ID number in the link URL path." so far, I only extracted your code for creating the Edit link into renderEditLink, which seems to work.  Is that enough, or do I need to loop through the stages to find one with the missing item?  

Also, I notice that some of the items have undefined field, but it is not clear why.

File: static/elements/chromedash-process-overview.js
Comment (jrobbins) [inline static/elements/chromedash-process-overview.js:163]: Rather than checking all the progress items in a stage, we should check all the progress item names in `action.prerequisites`.   That's because not all progress items are needed for a given action, and at the same time, actions can require progress items from earlier stages.  By looping over `action.prerequisites`, you would no longer need foundEmailItem because the prerequisites are named explicitly rather than depending on a naming convention.

**Review by jrobbins — CHANGES_REQUESTED** [review]

**Comment by jrobbins** [comment]
> Regarding your suggestion in #1801: "To make your dialog box have "Edit" links that jump straight to the field that needs to be edited, you will need to add logic that loops over the stages looking for the one that has the needed progress item. Then use that stage ID number in the link URL path." so far, I only extracted your code for creating the Edit link into renderEditLink, which seems to work. Is that enough, or do I need to loop through the stages to find one with the missing item?

I hope that my review comment on checkCompletion() explains what I had in mind.   For a given prerequisite name, you would need to loop through `stages` and `stage.progress_items` to find the progress item that has that name, and keep track of the stage ID that contains it.  When forming the URL of the link, you would put that stage ID in the URL path and `"id_" + field` in the anchor.

> Also, I notice that some of the items have undefined field, but it is not clear why.

Some progress items refer to work that happens outside of chromestatus, so there is no field to track whether that work has been done or not.  I don't think that any such field is a prerequisite for any action.

**Comment by dlaliberte** [comment]
Here is what the dialog looks like now:

image

Note that I changed the title to "Missing Prerequisites" and included the stage name along with each item.

**Comment by dlaliberte** [comment]
The "Edit fields" button doesn't make sense now, unless it goes to the page to edit all fields of all stages.  So I removed it for now.

File: static/elements/chromedash-process-overview.js
Comment (jrobbins) [inline static/elements/chromedash-process-overview.js:238]: This code looks ok, but the name `willUpdate` looks like you are saying that this is a `TODO`.  I don't see what calls this.

**Review by jrobbins — CHANGES_REQUESTED** [review]

File: static/elements/chromedash-process-overview.js
Comment (dlaliberte) [reply static/elements/chromedash-process-overview.js:238]: willUpdate is part of the lit element lifecycle, called before update(), which calls render().  So this functions the same as putting the code at the top of the render function, but it seems better in willUpdate, since it is not rendering anything.  See https://lit.dev/docs/components/lifecycle/#reactive-update-cycle-performing

Comment (jrobbins) [reply static/elements/chromedash-process-overview.js:238]: OK, I had not noticed that before.

**Review by jrobbins — APPROVED** [review]

File: static/elements/chromedash-process-overview.js
Comment (dlaliberte) [reply static/elements/chromedash-process-overview.js:166]: Missed this comment until now.  Can we require that all conversations are resolved before merging?

Comment (jrobbins) [reply static/elements/chromedash-process-overview.js:166]: I kind of like having it not required because it is an extra step.  Usually if there is something missed that is important, we can just do a small follow-up PR.

Comment (jrobbins) [reply static/elements/chromedash-process-overview.js:166]: The other part of the equation is that a reviewer should not give approval until after the review feedback that they feel is blocking has been addressed.  In this case, the logging is a minor issue, so I didn't hold my approval waiting for it, which saves one round of review.

Comment (dlaliberte) [reply static/elements/chromedash-process-overview.js:166]: OK.

**Comment by dlaliberte** [comment]
FYI, I'm experimenting with using Shoelace, first to replace the dialog I worked on here.  Here is what it looks like so far:

image

Note the large amount of whitespace, which is partly from the default style with 1em padding around the header and body, but also, because it is a padding, the space between the header and the body is at least 2em, with more padding for the ol.  I'm trying to fix this style via the Shoelace customization mechanism, but it is not working out yet.

**Pull request merged**
